From 223f021d5431a5287c631dd020f17dae98db6d36 Mon Sep 17 00:00:00 2001
From: Hendrik Leppkes <h.leppkes@gmail.com>
Date: Fri, 15 Mar 2013 16:02:37 +0100
Subject: [PATCH 14/63] Revert "mpegts: clear avprograms only for removed
 programs"

This reverts commit 806a66fd08a395486e763fb10588c5fd02e8d54e
and 4c41fc88df5c507e0f239b4c16f85573f7b49702.
---
 libavformat/mpegts.c | 27 ---------------------------
 1 file changed, 27 deletions(-)

diff --git a/libavformat/mpegts.c b/libavformat/mpegts.c
index a9a87fd..95bf95e 100644
--- a/libavformat/mpegts.c
+++ b/libavformat/mpegts.c
@@ -255,26 +255,10 @@ static struct Program * get_program(MpegTSContext *ts, unsigned int programid)
     return NULL;
 }
 
-static void clear_avprogram(MpegTSContext *ts, unsigned int programid)
-{
-    AVProgram *prg = NULL;
-    int i;
-
-    for (i = 0; i < ts->stream->nb_programs; i++)
-        if (ts->stream->programs[i]->id == programid) {
-            prg = ts->stream->programs[i];
-            break;
-        }
-    if (!prg)
-        return;
-    prg->nb_stream_indexes = 0;
-}
-
 static void clear_program(MpegTSContext *ts, unsigned int programid)
 {
     int i;
 
-    clear_avprogram(ts, programid);
     for (i = 0; i < ts->nb_prg; i++)
         if (ts->prg[i].id == programid) {
             ts->prg[i].nb_pids = 0;
@@ -2059,17 +2043,6 @@ static void pat_cb(MpegTSFilter *filter, const uint8_t *section, int section_len
             add_pid_to_pmt(ts, sid, pmt_pid);
         }
     }
-
-    if (sid < 0) {
-        int i,j;
-        for (j=0; j<ts->stream->nb_programs; j++) {
-            for (i = 0; i < ts->nb_prg; i++)
-                if (ts->prg[i].id == ts->stream->programs[j]->id)
-                    break;
-            if (i==ts->nb_prg && !ts->skip_clear)
-                clear_avprogram(ts, ts->stream->programs[j]->id);
-        }
-    }
 }
 
 static void sdt_cb(MpegTSFilter *filter, const uint8_t *section, int section_len)
-- 
2.5.0

